libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/Signaling/Signaler.cs

```diff
@@ -1,199 +1,255 @@
+            /// The SDP message content.
+            /// </summary>
+            public string Data;
+
+            public SdpMessage(string type, string data) : base(TypeFromString(type))
+            {
+                Data = data;
+            }
+
+            public static MessageType TypeFromString(string type)
+            {
+                if (type == "offer")
+                {
+                    return MessageType.Offer;
+                }
+                else if (type == "answer")
+                {
+                    return MessageType.Answer;
+                }
+                throw new ArgumentException($"Invalid SDP message type: \"{type}\"", "type");
+            }
+        }
+
+        [Serializable]
+        public class IceMessage : Message
+        {
+            /// <summary>
+            /// The ICE message content.
+            /// </summary>
+            public string Data;
+
+            /// <summary>
+            /// The data separator needed for proper ICE serialization.
+            /// </summary>
+            public string IceDataSeparator;
+
+            public IceMessage(string sdpMid, int sdpMlineIndex, string candidate) : base(MessageType.Ice)
+            {
+                IceDataSeparator = "|";
+                Data = string.Join(IceDataSeparator, new string[] { candidate, sdpMlineIndex.ToString(), sdpMid });
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously send a signaling message to the remote peer.
+        /// </summary>
+        /// <param name="message">The signaling message to send to the remote peer.</param>
+        /// <returns>
+        /// A <see cref="Task"/> object completed once the message has been sent,
+        /// but not necessarily delivered.
+        /// </returns>
+        public abstract Task SendMessageAsync(Message message);
+
+        #endregion
+
+
+        /// <summary>
+        /// Native <xref href="Microsoft.MixedReality.WebRTC.PeerConnection"/> object from the underlying
+        /// WebRTC C# library, available once the peer has been initialized.
+        /// </summary>
+        protected WebRTC.PeerConnection _nativePeer = null;
+
+        /// <summary>
+        /// Task queue used to defer actions to the main Unity app thread, which is the only thread
+        /// with access to Unity objects.
+        /// </summary>
+        protected ConcurrentQueue<Action> _mainThreadWorkQueue = new ConcurrentQueue<Action>();
+
+        /// <summary>
+        /// Callback fired from the <see cref="PeerConnection"/> when it finished
+        /// initializing, to subscribe to signaling-related events.
+        /// </summary>
+        /// <param name="peer">The peer connection to attach to</param>
+        public void OnPeerInitialized()
+        {
+            _nativePeer = PeerConnection.Peer;
+
+            // Register handlers for the SDP events
+            _nativePeer.IceCandidateReadytoSend += OnIceCandiateReadyToSend_Listener;
```

Comment:
There are a few "IceCandiate" that should be "Ice Candi**d**ate".

Reply:
meh 😞

Reply:
Fixed... in so many places 🤣